src/OrbitGl/CoreMath.h

```diff
@@ -29,14 +29,13 @@ using Color = gte::Vector4<unsigned char>;
 
 namespace orbit_gl {
 
+template <typename T>
 struct ClosedInterval {
   [[nodiscard]] bool Intersects(const ClosedInterval& closed_interval) const {
     return this->min <= closed_interval.max && this->max >= closed_interval.min;
   }
 
-  [[nodiscard]] bool Contains(float value) const {
```

Comment:
This struct would benefit from a static_assert that limits it to arithmetic types:

```cpp
static_assert(std::is_arithmetic_v<T>, "T needs to be an arithmetic type (integral or floating point type).");
```

Reply:
In the constructor?

Reply:
Just in the struct declaration - as if it was declaring a member variable.